### ISSUE TITLE
Support Alignment of DNA sequences with C API

### ISSUE THREAD

**Issue by adcosta17** [Daniel-Liu-c0deb0t/block-aligner#3]
I'd like to ask if it would be possible to support the alignment of DNA sequences using the C API? I would like to integrate `block-aligner` into an existing C/C++ tool but cannot use the exisiting C API to do alignments of DNA sequences.

**Comment by Daniel-Liu-c0deb0t** [comment]
Yup, this is something we will work on. I'm wondering if there is a better way to generate C bindings that is not just making multiple copies of a function to support Rust generics.

**Comment by d-cameron** [comment]
Are there any plans/progress on exposing the DNA sequence alignment in the C API? The Nuc API appears to be a subset of the AA API (no per-position scoring) and you seem to have the core alignment function macroed so it's a bit unclear what work is left other than mostly cut/paste of the non-macroed portion of the API. Was there any particular roadblock you hit or is it sufficiently straight-forward that someone much less familiar with the library could make a PR for the Nuc support?

**Comment by jianshu93** [comment]
Hi Both, Is there an update here. I have some large program for long reads genome assembly where the sse ksw was used but I want to try block aligner, it is nucleotide sequences. In the case of 10% bands block aligner should be be faster than: ksw_extz2_sse(0, ql, qs, tl, ts, 5, mat, gapo, gape, width, -1, end_bonus, flag, &ez);

Thanks,

Jianshu
